src/praisonai-agents/test_retry_fix.py

```
@@ -0,0 +1,86 @@
+        "Test error",
+        tool_name="test_tool",
+        agent_id="test_agent",
+        is_retryable=True
+    )
+    
+    # Set error_category as attribute (this is the fix)
+    err.error_category = "network"
+    
+    # Verify it's set correctly
+    assert err.error_category == "network"
+    assert err.is_retryable == True
```

Ruff (0.15.18)

[error] Avoid equality comparisons to `True`; use `err.is_retryable:` for truth checks

Replace with `err.is_retryable`

(E712)